src/commands/ui.rs

```diff
@@ -14,3 +37,99 @@ pub fn show_ui(options: &Options) -> anyhow::Result<()> {
         Err(ExitCode::new(1).into())
     }
 }
+
+pub fn ui_token(name: &str) -> anyhow::Result<()> {
+    println!("{}", generate_jwt(name)?);
+    Ok(())
+}
+
+fn generate_jwt(name: &str) -> anyhow::Result<String> {
+    // Replace this ES256/ECDH-ES implementation using raw ring
+    // with biscuit when the algorithms are supported in biscuit
+    let rng = rand::SystemRandom::new();
+
+    let data_dir = if name == "_localdev" {
+        data_dir()?.parent().unwrap().join("_localdev")
+    } else {
+        instance_data_dir(name)?
+    };
+
+    let mut buffer = Vec::new();
+    let mut f = fs::File::open(data_dir.join("edbjwskeys.pem"))?;
+    f.read_to_end(&mut buffer)?;
+    let pem = pem::parse(buffer)?;
+    let jws = signature::EcdsaKeyPair::from_pkcs8(
+        &signature::ECDSA_P256_SHA256_FIXED_SIGNING,
+        pem.contents.as_slice(),
+    )?;
+    let message = format!(
+        "{}.{}",
+        base64::encode_config(
+            b"{\"typ\":\"JWT\",\"alg\":\"ES256\"}",
+            base64::URL_SAFE_NO_PAD
+        ),
+        base64::encode_config(
+            b"{\"edgedb.server.any_role\":true}",
+            base64::URL_SAFE_NO_PAD
+        ),
+    );
+    let signature = jws.sign(&rng, message.as_bytes())?;
+    let signed_token = format!(
+        "{}.{}",
+        message,
+        base64::encode_config(signature, base64::URL_SAFE_NO_PAD),
+    );
+
+    let mut buffer = Vec::new();
+    let mut f = fs::File::open(data_dir.join("edbjwekeys.pem"))?;
+    f.read_to_end(&mut buffer)?;
+    let pem = pem::parse(buffer)?;
+    let jwe = signature::EcdsaKeyPair::from_pkcs8(
+        &signature::ECDSA_P256_SHA256_FIXED_SIGNING,
+        pem.contents.as_slice(),
+    )?;
+
+    let priv_key = agreement::EphemeralPrivateKey::generate(&agreement::ECDH_P256, &rng)?;
+    let pub_key =
+        agreement::UnparsedPublicKey::new(&agreement::ECDH_P256, jwe.public_key().as_ref());
+    let epk = priv_key.compute_public_key()?.as_ref().to_vec();
+    let cek = agreement::agree_ephemeral(priv_key, &pub_key, ExitCode::new(1), |key_material| {
```

Comment:
I don't think `ExitCode::new(1)` is a good idea here. We use `ExitCode` only in places where we print error description before returning `ExitCode`. So you could either use `anyhow::anyhow!(..)` here. But this perhaps has some cost on the success path. So I would use bare `()` and `.map_err(` with proper error formatting.